unittests/FEXLinuxTests/tests/signal/sigtest_siginfo.64.cpp

```diff
@@ -0,0 +1,41 @@
+#include <catch2/catch.hpp>
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <cstdint>
+
+__attribute__((naked, nocf_check))
+static void CauseInt() {
+  __asm volatile(R"(
+  int $1;
+  ret; # For RIP modification
+  )");
+}
+
+static uint64_t EXPECTED_RIP = reinterpret_cast<uint64_t>(&CauseInt);
+constexpr int EXPECTED_SIGNAL = SIGSEGV;
+
+static void Handler(int Signal, siginfo_t *siginfo, void *context) {
+  ucontext_t* _context = (ucontext_t*)context;
+
+  if (_context->uc_mcontext.gregs[REG_RIP] == EXPECTED_RIP &&
+      Signal == EXPECTED_SIGNAL) {
+    _context->uc_mcontext.gregs[REG_RIP] += 2;
+  }
+  else {
+    exit(-1);
+  }
+}
+
+TEST_CASE("siginfo") {
+  // On x86-64, the signal handler receives siginfo even if SA_SIGINFO isn't set.
+  // This flag is effectively a no-op, not changing behaviour.
+  struct sigaction act{};
+  act.sa_sigaction = Handler;
+  act.sa_flags = 0;
+  sigaction(SIGSEGV, &act, nullptr);
+
+  CauseInt();
```

Comment:
How about adding a variant here with SA_SIGINFO set to check that the flag does have an effect on 32-bit?

Reply:
That's the next step after this PR. We don't handle some of the 32-bit subtleties around this yet.
32-bit has four combinations that we need to support.
- sigaction + SA_SIGINFO
- sigaction - SA_SIGINFO
- rt_sigaction + SA_SIGINFO
- rt_sigaction - SA_SIGINFO

I wanted this PR to stay about 64-bit so I can solve the 32-bit side independently.